dask/array/core.py

```diff
@@ -240,6 +241,198 @@ def slices_from_chunks(chunks):
     return list(product(*slices))
 
 
+def subset_to_blocks(
+    data,
+    *,
+    coords: tuple[tuple[int]],
+    new_chunks: tuple[int] = (1, 1),
+    token="",
+):
+    token = f"{token + '-' if token else token}-subset-blocks-{tokenize(data, *coords)}"
+    keys = data._key_array[..., *coords, :]
+    graph = {}
+    for flatidx, idx in enumerate(np.ndindex(keys.shape[:-1])):
+        graph[(token, *idx[:-1], flatidx)] = tuple(keys[*idx, :])
+
+    ndim = len(coords)
+    return Array(
+        HighLevelGraph.from_collections(token, graph, dependencies=[data]),
+        token,
+        chunks=(*data._chunks[:-ndim], new_chunks),
+        meta=data,
+    )
+
+
+# Copied from Xarray
+def inverse_permutation(indices: np.ndarray, N: int | None = None) -> np.ndarray:
+    """Return indices for an inverse permutation.
+
+    Parameters
+    ----------
+    indices : 1D np.ndarray with dtype=int
+        Integer positions to assign elements to.
+    N : int, optional
+        Size of the array
+
+    Returns
+    -------
+    inverse_permutation : 1D np.ndarray with dtype=int
+        Integer indices to take from the original array to create the
+        permutation.
+    """
+    if N is None:
+        N = len(indices)
+    # use intp instead of int64 because of windows :(
+    inverse_permutation = np.full(N, -1, dtype=np.intp)
+    inverse_permutation[indices] = np.arange(len(indices), dtype=np.intp)
+    return inverse_permutation
+
+
+def interp_helper(
+    func: Callable[np.ndarray, ...],
+    data,
+    *,
+    x: tuple[np.ndarray, ...],
+    new_x: tuple[np.ndarray, ...],
```

Comment:
A nicer signature would be interleaved `x0, new_x0, x1, new_x1`

Reply:
This is only used internally, correct? If yes, feel free to use the signature that is easier to work with